psi4/driver/procrouting/proc.py

```diff
@@ -3165,36 +3165,56 @@ def run_cc_property(name, **kwargs):
+        if name.startswith("eom"):
+            gs_h = 0
+            for h, i in enumerate(ccwfn.soccpi()):
+                if i % 2:
+                    gs_h = gs_h ^ h
+            ct = ccwfn.molecule().point_group().char_table()
+            total_h_lbl = ct.gamma(0).symbol()
+            gs_h_lbl = ct.gamma(gs_h).symbol()
+            set_of_names.update({title + " ROOT 0 {}", "CC ROOT 0 {}",
+                                 f"{title} ROOT 0 {{}} - {total_h_lbl} TRANSITION",
+                                      f"CC ROOT 0 {{}} - {total_h_lbl} TRANSITION",
+                                 f"{title} ROOT 0 ({gs_h_lbl}) {{}}", f"CC ROOT 0 ({gs_h_lbl}) {{}}"})
+        oe.set_names(set_of_names)
         oe.compute()
-        # call oe prop for each ES density
+        print(ccwfn.variables())
+        # ==> OEProp for Excited States <==
         if name.startswith('eom'):
-            # copy GS CC DIP/QUAD ... to CC ROOT 0 DIP/QUAD ... if we are doing multiple roots
-            if 'DIPOLE' in one:
-                core.set_variable("CC ROOT 0 DIPOLE", core.variable("CC DIPOLE"))
-                # core.set_variable("CC ROOT n DIPOLE", core.variable("CC DIPOLE"))  # P::e CCENERGY
-            if 'QUADRUPOLE' in one:
-                core.set_variable("CC ROOT 0 QUADRUPOLE", core.variable("CC QUADRUPOLE"))
-                # core.set_variable("CC ROOT n QUADRUPOLE", core.variable("CC QUADRUPOLE"))  # P::e CCENERGY
-
-            n_root = sum(core.get_global_option("ROOTS_PER_IRREP"))
-            for rn in range(n_root):
-                oe.set_title("CC ROOT {}".format(rn + 1))
-                Da = ccwfn.variable("CC ROOT {} Da".format(rn + 1))
-                oe.set_Da_so(Da)
-                if core.get_global_option("REFERENCE") == "UHF":
-                    Db = ccwfn.variable("CC ROOT {} Db".format(rn + 1))
-                    oe.set_Db_so(Db)
-                oe.compute()
+            n_root_pi = core.get_global_option("ROOTS_PER_IRREP")
+            for h in range(ccwfn.nirrep()):
+                root_h_lbl = ct.gamma(h).symbol()
+                trans_h_lbl = ct.gamma(h ^ gs_h).symbol()
+                # Don't forget to count the ground state!
+                for i in range(n_root_pi[h]):
+                    if h == gs_h: i += 1
+                    root_title = title + f" ROOT {i} ({root_h_lbl})"
+                    oe.set_title(root_title)
+                    total_idx = ccwfn.total_index(i, h)
+                    set_of_names = {f"{title} ROOT {total_idx} {{}}", f"CC ROOT {total_idx} {{}}",
+                                    f"{title} ROOT {total_idx} {{}} - {trans_h_lbl} TRANSITION",
+                                         f"CC ROOT {total_idx} {{}} - {trans_h_lbl} TRANSITION",
+                                    f"{title} ROOT {i} ({root_h_lbl}) {{}}", f"CC ROOT {i} ({root_h_lbl}) {{}}"}
+                    oe.set_names(set_of_names)
+                    Da = ccwfn.variable(root_title + " Da")
```

Comment:
Are the density qcvars only for intracode communication, or are there definitely reasons to expose them to the user? There's ordering and size considerations against keeping them around.

Reply:
Not for intracode communication. I didn't see including them as controversial, given that we keep ground-state densities around.

Reply:
I'm all for keeping them around, just not in qcvars, since those get exported and printed thoughtlessly. I'd favor a different member data on Wfn. Others thoughts?